packages/storage/src/Storage.ts

```
@@ -159,14 +169,47 @@ export class Storage {
 		return this._config;
 	}
 
+	private getCancellableTokenSource(): CancelTokenSource {
+		return axios.CancelToken.source();
+	}
+
+	private updateRequestToBeCancellable(
+		request: Promise<any>,
+		cancelTokenSource: CancelTokenSource
+	) {
+		this._cancelTokenSourceMap.set(request, cancelTokenSource);
+	}
+
+	/**
+	 * Cancels an inflight request
+	 *
+	 * @param {Promise<any>} request - The request to cancel
+	 * @param {Object} [config] - { provider: string }
+	 * @param {string} [message] - A message to include in the cancelation exception
+	 */
+	public cancel(request: Promise<any>, config?, message?: string) {
+		const { provider = DEFAULT_PROVIDER } = config || {};
+		const prov = this._pluggables.find(
+			pluggable => pluggable.getProviderName() === provider
+		);
+		if (prov === undefined) {
+			logger.debug('No plugin found with providerName', provider);
+			Promise.reject('No plugin found in Storage for the provider');
```

in my opinion, we should throw a custom error type, that could be used later to differentiate it from another error.

---

ok, now I see that you're just replicating some logic from siblings methods. Please discard this message.